[UI] Polish UI and Bugfixes

alexis-ld opened the pull request:
**Component**:
ui, volumes, nodes, sidebar

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

- [Improvement]We want to keep the time selector and advanced metrics button always visible in the tabs when we scroll down.
- [Improvement] We can still improve the Volume Page Metrics tab, in order to fully use the empty space (Please refer to the Node Page Metrics tab, maybe we should generalize the common styles).
- [Improvement] Add a separation line in the volume table between Header and content rows
- [Bug] In Volume Page, the Node Name should come after Volume Name
- [Improvement] Rename the overview page to Alerts in the sidebar
- [Bug] When sorting by size in the volume list, we should take into consideration Unknown.

**Acceptance criteria**: 
- The time selector stays visible on metrics tab when the user scrolls down
- The Charts are now taking the available space in metrics tabs and the layout is more coherent between the Volumes and Nodes pages
- The node column is now after the name column on Volumes table
- Overview is renamed to alerts in the sidebar
- Sorting by size on Volumes table is now working as expected


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2917

<!-- If you want to refer to an issue while not closing it, use:


-->

Commit:
ui/volumes/nodes: Isolate metrics actions container to avoid code redundancy

Commit:
ui/volumes/nodes: Make metrics tab actions sticky on top

Refs: #2917

Commit:
ui/volumes: Add missing border on Volumes table header

Refs: #2917

Commit:
ui/volumes: Make Node name come after Volume name in Volumes table

Refs: #2917

Commit:
ui/volumes: Fix volumes size sorting

In case of an undefined value nothing was returned in the sort function making the sorting unpredictable

Refs: #2917

Commit:
ui/sidebar: Rename Monitoring to Alerts

Refs: #2917

Commit:
ui/charts: Unifying charts dimensions and layout

Common components to handle charts layout through Volumes and Nodes page

Custom hook to define chart dimension dynamically to use through the app

Refs: #2917

bert-e commented:
# Hello alexis-ld,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2917-polish-ui` into
`development/2.6`
* [w/2.7/improvement/2917-polish-ui](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/2917-polish-ui) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

thomasdanan reviewed (changes requested):
When going to metric tab, charts are first displayed with a small size and few seconds after resize to fill the entire space. This is a bit annoying.

Also, the Volume Name in Volume Page Overview tab and Node Name in Node Page overview tab is too much aligned on the top. IN the previous version there was a little margin which is prettier in my opinion.

All the rest is okay.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @thomasdanan

Commit:
ui/volumes/nodes: Add missing padding on overview tabs

Refs: #2917

Commit:
ui/volumes/nodes: Wait for width before chart display

Added graphWidth as a condition to display the chart to avoid flashing on unsized chart

Refs: #2917

alexis-ld commented:
@thomasdanan Indeed I missed the padding on the overview tab, sorry about that. Should be fixed now.
Also the flash of unsized charts should be fixed as well.

ChengYanJin commented on ui/src/components/VolumeListTable.js:
```diff
@@ -57,6 +57,11 @@ const VolumeListContainer = styled.div`
       height: 10px;
     }
+
+    thead tr[role='row'] {
+      border-bottom: 1px solid ${(props) => props.theme.brand.border};
+    }
```
You may want to take a quick look at this [commit] (https://github.com/scality/metalk8s/pull/2938/commits/9163f9799b4387abcc2e0f59f72eaeb9ed7e0587) which I replaced all the HTML tags by `<div/>`

You may want to rebase on my branch then... 

Update:

I addressed on my PR https://github.com/scality/metalk8s/pull/2938/files

ChengYanJin reviewed (changes requested):
Hi @alexis-ld, 
Do we consider sticking the alert selection in the Alert tab and the create volume button in the Volume tab too?  Just like what you did for the Metrics tab.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

* @thomasdanan

Commit:
ui/volumes/nodes: Make top action buttons sticky on Volumes and Nodes tabs

Refs: #2917

Commit:
ui/volumes/nodes: Disable tab scrolling

Disabled tab scrolling in tabs that contains a scrollable table in order to leverage table scrolling and sticky headers

Refs: #2917

ChengYanJin reviewed (approved):
LGTM!

alexis-ld commented:
/approve

ChengYanJin commented:
coucou

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **approve**

ChengYanJin commented:
/status

bert-e commented:
# Status





*Status report is not available.*






*The following options are set:* **approve**

ghost commented:
This PR is blocked because development/2.7 does not contains the HEAD of development/2.6, 2 commits are missing, and so Bert-E refuse to forward port. This need to be fixed first.

This should be more explicit on the PR anyway, i created a ticket for that https://scality.atlassian.net/browse/BERTE-537

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/56a1699e539717c461dca7ec844fd54ad9a61667) did not succeed in branch w/2.7/improvement/2917-polish-ui.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`

* :heavy_check_mark: `development/2.7`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye alexis-ld.
